### DISCUSSION TITLE
How can I add 'None' as a filter option to return records where filtered column is null?

### DISCUSSION THREAD

**Discussion by LemonPie22** [filamentphp/filament#9723] [Help]
I have a `parts` resource with various filters on the table:

```
->filters(
    [
        Tables\Filters\TrashedFilter::make(),
        Tables\Filters\SelectFilter::make('category')
            ->relationship('partCategory', 'name'),
        Tables\Filters\SelectFilter::make('part_type_id')
            ->relationship('partType', 'name')
            ->label('Type'),
        Tables\Filters\SelectFilter::make('make_id')
            ->relationship('make', 'name')
            ->label('Make'),
        Tables\Filters\SelectFilter::make('model_id')
            ->relationship('partModel', 'name')
            ->label('Model'),
        Tables\Filters\SelectFilter::make('brand_id')
            ->relationship('brand', 'name')
            ->label('Brand'),
```

The result is several filters on the parts table:
<img width="1291" alt="Screenshot 2023-11-17 at 22 34 27" src="https://github.com/filamentphp/filament/assets/107488019/96a04b89-2215-4fd4-8500-86a13bc65540">

When the user chooses to filter on `type`, for example, they get a select as expected:
<img width="440" alt="Screenshot 2023-11-17 at 22 37 35" src="https://github.com/filamentphp/filament/assets/107488019/5e08c19d-cff6-42c7-b0e0-14ecc1a0bade">

But what if `type` is nullable? Then there will be parts without a type. How can I offer the user the ability to filter on them?

I tried making the `type` filter nullable but that doesn't work since that's only for ternary filters (I believe).

So how can I add "None" to the select and return parts where `type` is null?

**Comment by LemonPie22** [comment]
I can achieve this using 2 separate filters, one ternary and one select, like this:

```
Tables\Filters\TernaryFilter::make('template_id')
    ->label('Has template')
    ->nullable(),
Tables\Filters\SelectFilter::make('template')
    ->attribute('template_id')
    ->relationship('template', 'name'),
```
But I'd ideally like to use a single filter with 'None' in the list.

Any ideas?

**Reply by softwaretechniker** [reply]
Idea 1: Write a custom NullableSelectFilter
Idea 2: Trying to mess around with the relationship within the model (using a 2nd relation where some query is added)
Idea 3: Contributing directly to Filament and hoping pull-request will be accepted
Idea 4: Checking if it's possible to add it by customizing the component's view template or simply with ->getSearchResultsUsing()
Idea 5: Hoping the feature will be implemented in Filament 4, 5, 6 or whatever in the next years

**Reply by softwaretechniker** [reply]
Edit: Idea 4 with getSearchResultsUsing() won't work at the moment: There's a bug with it: [discussions/14215](https://github.com/filamentphp/filament/discussions/14215)
